enclave/core/sgx/init.c

```diff
@@ -60,8 +60,25 @@ static oe_result_t _eeid_patch_memory()
     {
         uint8_t* enclave_base = (uint8_t*)__oe_get_enclave_base();
         uint8_t* heap_base = (uint8_t*)__oe_get_heap_base();
-        oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
-        oe_eeid = (oe_eeid_t*)(enclave_base + marker->offset);
+        const oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
```

Comment:
@wintersteiger @bodzhang 
We don't measure the contents of the heap pages currently. Therefore the host could initialize it to what-ever value it wants to. Do we have special logic to measure the contents of the marker page even though it is the first heap page?

Reply:
The marker page can't be measured at signing time as we don't know the offset, but I suppose we could measure it at enclave creation time and update the verifier accordingly. There are a number of checks in `init.c` to make sure it points somewhere sensible, but I don't think it makes sure that it's on a measured page. @bodzhang what's your opinion, should the marker be measured?

Reply:
This PR is a stopgap fix before the suggestions proposed in #3269, correct? In that updated design, there is a fixed EEID page in the enclave layout, which precludes the retargeting attack. That would be my preference, but the current fix is fine in the meantime.

Reply:
That's correct, this is just a fix for the current design, the new one won't won't use a marker page.